core/src/test/java/org/apache/iceberg/TestMetadataUpdateParser.java

```diff
@@ -738,6 +740,40 @@ public void testSetLocationToJson() {
         "Remove properties should serialize to the correct JSON value", expected, actual);
   }
 
+  /** SetStatistics */
+  @Test
+  public void testSetStatistics() {
+    String json =
+        "{\"action\":\"set-statistics\",\"snapshot-id\":42,\"path\":\"s3://bucket/warehouse/stats.puffin\","
+            + "\"file-size-in-bytes\":124,\"file-footer-size-in-bytes\":27,\"blob-metadata\":[{\"type\":\"boring-type\",\"source-snapshot-id\":42,\"source-snapshot-sequence-number\":2,\"fields\":[1],\"properties\":{\"prop-key\":\"prop-value\"}}]}";
+    MetadataUpdate expected =
+        new MetadataUpdate.SetStatistics(
+            42,
+            new GenericStatisticsFile(
+                42,
+                "s3://bucket/warehouse/stats.puffin",
+                124,
+                27,
+                ImmutableList.of(
+                    new GenericBlobMetadata(
+                        "boring-type",
+                        42,
+                        2,
+                        ImmutableList.of(1),
+                        ImmutableMap.of("prop-key", "prop-value")))));
+    assertEquals(
+        MetadataUpdateParser.SET_STATISTICS, expected, MetadataUpdateParser.fromJson(json));
+  }
+
+  /** RemoveStatistics */
```

Comment:
Can you remove these comments? This context is clear from the test name so these aren't needed.